ingester/src/data.rs

```diff
@@ -79,48 +93,197 @@ pub struct IngesterData {
+    /// The content of each SequenceData will get changed when more namespaces and tables
+    /// get ingested.
     pub(crate) sequencers: BTreeMap<SequencerId, SequencerData>,
+    /// Executor for running queries and compacting and persisting
+    pub(crate) exec: Executor,
 }
 
 impl IngesterData {
     /// Store the write or delete in the in memory buffer. Deletes will
     /// be written into the catalog before getting stored in the buffer.
     /// Any writes that create new IOx partitions will have those records
-    /// created in the catalog before putting into the buffer.
+    /// created in the catalog before putting into the buffer. Writes will
+    /// get logged in the lifecycle manager. If it indicates ingest should
+    /// be paused, this function will return true.
     pub async fn buffer_operation(
         &self,
         sequencer_id: SequencerId,
         dml_operation: DmlOperation,
-    ) -> Result<()> {
+        lifecycle_manager: &LifecycleManager,
+    ) -> Result<bool> {
         let sequencer_data = self
             .sequencers
             .get(&sequencer_id)
             .context(SequencerNotFoundSnafu { sequencer_id })?;
         sequencer_data
-            .buffer_operation(dml_operation, sequencer_id, self.catalog.as_ref())
+            .buffer_operation(
+                dml_operation,
+                sequencer_id,
+                self.catalog.as_ref(),
+                lifecycle_manager,
+            )
             .await
     }
 }
 
 /// The Persister has a single function that will persist a given partition Id. It is expected
 /// that the persist function will retry forever until it succeeds.
 #[async_trait]
-pub(crate) trait Persister: Send + Sync + 'static {
+pub trait Persister: Send + Sync + 'static {
+    /// Persits the partition ID. Will retry forever until it succeeds.
     async fn persist(&self, partition_id: PartitionId);
 }
 
 #[async_trait]
 impl Persister for IngesterData {
-    async fn persist(&self, _partition_id: PartitionId) {
-        // lookup the TableData
-        // let persisting_batch = table_data.create_persisting_batch(partition.partition_key);
-        // do the persist with this persisting batch
-        // update the catalog
-        // table_data.clear_persisting_batch() (behind the scenes this will remove the persisting batch
-        // and if the partition is empty, remove it from the map in table_data)
+    async fn persist(&self, partition_id: PartitionId) {
+        let retry_time = Duration::from_secs(1);
+
+        let mut repos = self.catalog.repositories().await;
+
+        // lookup the partition_info from the catalog
+        let partition_info: Option<PartitionInfo> = loop {
+            match repos.partitions().partition_info_by_id(partition_id).await {
+                Ok(p) => break p,
+                Err(e) => {
+                    error!(%e, "error getting partition_info_by_id");
```

Comment:
Should this be logged as an error? An error-level log should be reserved for when something significant has happened that's unrecoverable and an operator need to intervene somehow. But given there is a retry here I don't think that's warranted. What are the circumstances under which this lookup fails? Perhaps `warn` or even `info` might be a better level. 

```suggestion
                    warn!(%e, ?partition_id, "persist partition ID lookup failed");
```